sub/sd_ass.c

```diff
@@ -289,45 +289,36 @@ static int init(struct sd *sd)
 }
 
 // Check if subtitle has events that would cause it to be animated inside {}
-static bool is_animated(char *s)
+static bool is_animated(const char *str)
 {
-    bool in_tag = false;
-    bool valid_event = false;
-    bool valid_tag = false;
-    while (*s) {
-        if (!in_tag && s[0] == '{')
-            in_tag = true;
-        if (s[0] == '\\') {
-            s++;
-            if (!s[0])
-                break;
-            if (s[0] == 'k' || s[0] == 'K' || s[0] == 't') {
-                valid_event = true;
-                continue;
-            // just bruteforce the multi-letter ones
-            } else if (s[0] == 'f') {
-                if (!strncmp(s, "fad", 3)) {
-                    valid_event = true;
-                    continue;
-                }
-            } else if (s[0] == 'm') {
-                if (!strncmp(s, "move", 4)) {
-                    valid_event = true;
-                    continue;
-                }
+    if (str[0] == '{')
+        goto found;
+    while ((str = strchr(str, '{'))) {
+        ++str;
+        if (str[-2] == '\\')
+            continue;
+found:;
+        const char *end = strchr(str, '}');
```

Comment:
Why not `strlen` at the beginning and use `memchr` to find `{` and `}` in iterations? I suppose `memchr` has better performance.

Reply:
An optimized `strchr` will be just as fast as a `strlen`, and avoids an extra pass here.

Reply:
Not really. We don't need len. To compute len, we have to scan whole string and find NUL char. We only need to scan to first `{`, which can be very small portion of the string. `strlen` is kinda specialized version of `strchr(s, '\0')`. If we had length we could use `memchr`, but we don't and calculating it is way more work.